### PR TITLE
Add robot-full-permission proposal

### PR DESCRIPTION
**Pull request by AllForNothing**
Original issue https://github.com/goharbor/harbor/issues/8723

1.Expose all the robot permissions on the Harbor UI for users to choose freely

### CODE REVIEW

**Review by Vad1mo — CHANGES_REQUESTED** [review]
This looks good so far.
I Have the fear that this feature will overwhelm a typical Harbor day to day user and use case.
As a suggestion, can we break up this UI change into two parts?

1.  Keep the current selection with the typical dropdown is present where the use can select the common permission. 

2. Add an expert mode where the user is presented with the matrix and can select. individual permissions. What is currently displayed in the screenshots.

**Comment by AllForNothing** [comment]
@Vad1mo The screenshots are out of date, I will update them later.
I've refactored it based on the designer's suggestions and now it looks similar to your suggestions. And I will demo the new UI in today's community meeting.

**Comment by OrlinVasilev** [comment]
that would be so cool with the expert mode!!!

**Review by wy65701436 — APPROVED** [review]
lgtm

File: proposals/new/robot-full-permission.md [hunk 1]
```diff
@@ -0,0 +1,583 @@
...
+106. post   /users                                          resource:  user     |   action: create
+107. get    /users/current                                  not allowed
+108. get    /users/search                                   default
+109. get    /users/{user_id}                                resource:  user     |   action: read
+110. put    /users/{user_id}                                resource:  user     |   action: update
+111. delete /users/{user_id}                                resource:  user     |   action: delete
+112. put    /users/{user_id}/sysadmin                       resource:  user     |   action: update
+113. put    /users/{user_id}/password                       resource:  user     |   action: update
+114. get    /users/current/permissions                      default
+115. put    /users/{user_id}/cli_secret                     resource:  user     |   action: update
```
Comment (YangJiao0817) [inline proposals/new/robot-full-permission.md:395]: The user API should be disabled。

**Review by YangJiao0817 — CHANGES_REQUESTED** [review]

**Review by YangJiao0817 — APPROVED** [review]

**Pull request merged**